packages/core/src/schema/types.ts

```diff
@@ -0,0 +1,25 @@
+import { apiKeys } from '$core/schema/models/apiKeys'
+import { commits } from '$core/schema/models/commits'
+import { documentVersions } from '$core/schema/models/documentVersions'
+import { memberships } from '$core/schema/models/memberships'
+import { projects } from '$core/schema/models/projects'
+import { providerApiKeys } from '$core/schema/models/providerApiKeys'
+import { sessions } from '$core/schema/models/sessions'
+import { users } from '$core/schema/models/users'
+import { workspaces } from '$core/schema/models/workspaces'
+import { type InferSelectModel } from 'drizzle-orm'
+
+// Model types are out of schema files to be able to share with NextJS webpack bundler
+// otherwise, it will throw an error.
+export type Workspace = InferSelectModel<typeof workspaces>
+export type User = InferSelectModel<typeof users>
+export type SafeUser = Pick<User, 'id' | 'name' | 'email'>
+export type Session = InferSelectModel<typeof sessions> & {
+  user: User
+}
+export type Membership = InferSelectModel<typeof memberships>
+export type ProviderApiKey = InferSelectModel<typeof providerApiKeys>
+export type ApiKey = InferSelectModel<typeof apiKeys>
+export type Commit = InferSelectModel<typeof commits>
+export type DocumentVersion = InferSelectModel<typeof documentVersions>
+export type Project = InferSelectModel<typeof projects>
```

Comment:
hmm you shouldn't need this if you only import types from a schema file it does not get bundled client-side and everything works fine, problem starts when you import anything other than types from a schema file

Reply:
I sick of dealing with this problem. I think is clear if we put all schema types in one file that we now we can call from anywhere